File: apps/web/src/lib/schemas/audit.ts
import { z } from "zod";
import { paginationSchema } from "./common";

export const auditEventSchema = z.object({
  id: z.string(),
  msme_id: z.string().nullable(),
  event_type: z.string(),
  actor: z.string(),
  request_id: z.string().nullable(),
  created_at: z.string(),
  metadata: z.record(z.unknown())
});

export const auditListResponseSchema = z.object({
  items: z.array(auditEventSchema),
  pagination: paginationSchema
});

export type AuditEvent = z.infer<typeof auditEventSchema>;
export type AuditListResponse = z.infer<typeof auditListResponseSchema>;

File: apps/api/app/schemas/audit.py
from datetime import datetime

from pydantic import BaseModel

from app.schemas.common import Pagination


class AuditEventSchema(BaseModel):
    id: str
    msme_id: str | None
    event_type: str
    actor: str
    request_id: str | None
    created_at: datetime
    metadata: dict


class AuditListResponse(BaseModel):
    items: list[AuditEventSchema]
    pagination: Pagination
